fix: prevent double pagination and bad count value

nicolastoulemont opened the pull request:
Fixes https://github.com/askorama/orama/issues/730 by removing the double pagination which upperbounded the count value and caused some missed documents in the documents fetching later in the execution path.

nicolastoulemont commented:
@micheleriva Ready for review :)
If the PR is good for you, I'm not claiming anything bug bounty fyi

rajeshj11 commented:
> @micheleriva Ready for review :) If the PR is good for you, I'm not claiming anything bug bounty FYI

this is my first commit. I have missed the test cases. which I have later added.
https://github.com/askorama/orama/pull/731/commits/2200f15934babc3b09a6ecdd66622be7063510a0

nicolastoulemont commented:
> > @micheleriva Ready for review :) If the PR is good for you, I'm not claiming anything bug bounty FYI
> 
> this is my first commit. I have missed the test cases. which I have later added. [2200f15](https://github.com/askorama/orama/commit/2200f15934babc3b09a6ecdd66622be7063510a0)

For the record, I didn't take your work as much as I implemented the changes I suggested on [slack](https://orama-community.slack.com/archives/C0445CLH631/p1717166829824649) since your PR tests weren't ready, hoping for a quicker merge :)

Happy to close my PR in favor of your tho, especially since you participate in the BB program which I have no interests in myself and fixed your tests. I will add another comment on your PR to help you finish it
